tests/TensorAlgebra_Jobs.cpp

```diff
@@ -0,0 +1,1913 @@
+#include "einsums/Jobs.hpp"
+#include "einsums/LinearAlgebra.hpp"
+#include "einsums/STL.hpp"
+#include "einsums/Sort.hpp"
+#include "einsums/State.hpp"
+#include "einsums/Tensor.hpp"
+#include "einsums/TensorAlgebra.hpp"
+#include "einsums/Utilities.hpp"
+
+#include <H5Fpublic.h>
+#include <catch2/catch_all.hpp>
+#include <chrono>
+#include <complex>
+#include <cstdio>
+#include <thread>
+#include <type_traits>
+
+TEST_CASE("Identity Tensor Locks", "[jobs]") {
+    using namespace einsums;
+    using namespace einsums::tensor_algebra;
+    using namespace einsums::jobs;
+
+    Tensor<double, 2> *_I = new Tensor<double, 2>(create_identity_tensor("I", 3, 3));
```

Comment:
I'm making a small fix. I don't think letting the resource manage a foreign pointer will be good for memory leaks, so I changed it so that the resource creates its own instance. Also, it needs to be a pointer because while the heap is shared between threads, each thread has its own stack. By allocating a non-pointer, it is put on the stack and can not be shared between threads.

Reply:
Isn't there also an issue that Resource uses a shared_ptr internally but what is passed in isn't a shared_ptr? Nothing will stop me from deleting _I and invalidating the Resource.

Reply:
What do you mean by create its own instance? If I is large wouldn't that be wasteful?

Reply:
That's what I mean. As I have it now on my local machine, Resource will create a new instance by calling make_shared and passing constructor arguments. I can post what I have, but I'm running into some issues with segfaults and deadlocks.